pkg/cmd/agent/view.go

```diff
@@ -0,0 +1,48 @@
+package agent
+
+import (
+	"fmt"
+
+	"github.com/buildkite/cli/v3/internal/printer"
+	"github.com/buildkite/cli/v3/pkg/cmd/factory"
+	"github.com/buildkite/cli/v3/pkg/cmd/validation"
+	"github.com/spf13/cobra"
+)
+
+func NewCmdAgentView(f *factory.Factory) *cobra.Command {
+	var id string
+	var output string
+	cmd := cobra.Command{
+		Use:   "view",
+		Args:  cobra.NoArgs,
+		Short: "View an agent",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			id, _ = cmd.Flags().GetString("id")
+			output, _ = cmd.Flags().GetString("output")
+			if id == "" {
+				return fmt.Errorf("id is required")
+			}
+			if output == "" {
+				output = "json"
+			}
+			err := validation.ValidateUUID(id)
+			if err != nil {
+				return err
+			}
+			agent, _, err := f.RestAPIClient.Agents.Get(f.Config.Organization, id)
+			if err != nil {
+				return err
+			}
+			data, err := printer.PrintOutput(printer.Output(output), agent)
+			if err != nil {
+				return err
+			}
+			fmt.Println(data)
+			return nil
+		},
+	}
+
+	cmd.Flags().String("id", "", "The ID of the agent")
```

Comment:
I don't think this should be a flag, it should be an arg? eg [`gh issue view <num>`](https://primer.style/cli/foundations/language)

Reply:
Depends entirely on which CLI you're trying to replicate; in `aws` it's a flag.